File: hooks/push-usage.py
import json
import urllib.request
import sys
import os
from pathlib import Path

SUPABASE_EDGE_URL = "https://uskyccnonglqfvjsqkad.supabase.co/functions/v1/push-usage"
CREDS_FILE = Path.home() / ".claude" / ".credentials.json"

def main():
    member_name = os.environ.get("CLAUDE_MEMBER_NAME", "unknown")

    if not CREDS_FILE.exists():
        print("credentials file not found", file=sys.stderr)
        return

    creds = json.loads(CREDS_FILE.read_text())
    token = creds["claudeAiOauth"]["accessToken"]

    req = urllib.request.Request(
        "https://api.anthropic.com/api/oauth/usage",
        headers={
            "Authorization": f"Bearer {token}",
            "anthropic-client-name": "claude-code",
            "anthropic-version": "2023-06-01",
        }
    )
    try:
        with urllib.request.urlopen(req) as res:
            usage = json.loads(res.read())
    except Exception as e:
        print(f"failed to fetch usage: {e}", file=sys.stderr)
        return

    fh = usage.get("five_hour") or {}
    sd = usage.get("seven_day") or {}

    payload = json.dumps({
        "member_name":            member_name,
        "five_hour_utilization":  fh.get("utilization"),
        "five_hour_resets_at":    fh.get("resets_at"),
        "seven_day_utilization":  sd.get("utilization"),
        "seven_day_resets_at":    sd.get("resets_at"),
    }).encode()

    req = urllib.request.Request(
        SUPABASE_EDGE_URL,
        data=payload,
        headers={"Content-Type": "application/json"},
        method="POST"
    )
    try:
        urllib.request.urlopen(req)
        print(f"usage pushed for {member_name}")
    except Exception as e:
        print(f"failed to push: {e}", file=sys.stderr)

if __name__ == "__main__":
    main()

File: hooks/push-usage.js
const https = require('https')
const fs = require('fs')
const path = require('path')
const os = require('os')

const SUPABASE_EDGE_URL = 'https://uskyccnonglqfvjsqkad.supabase.co/functions/v1/push-usage'
const CREDS_FILE = path.join(os.homedir(), '.claude', '.credentials.json')

function request(url, options, body) {
  return new Promise((resolve, reject) => {
    const u = new URL(url)
    const req = https.request({
      hostname: u.hostname,
      path: u.pathname + u.search,
      method: options.method || 'GET',
      headers: options.headers || {},
    }, res => {
      let data = ''
      res.on('data', chunk => data += chunk)
      res.on('end', () => {
        if (res.statusCode >= 400) reject(new Error(`HTTP ${res.statusCode}: ${data}`))
        else resolve(data)
      })
    })
    req.on('error', reject)
    if (body) req.write(body)
    req.end()
  })
}

async function main() {
  const memberName = process.env.CLAUDE_MEMBER_NAME || os.userInfo().username

  if (!fs.existsSync(CREDS_FILE)) {
    console.error('credentials file not found:', CREDS_FILE)
    return
  }

  const creds = JSON.parse(fs.readFileSync(CREDS_FILE, 'utf-8'))
  const token = creds?.claudeAiOauth?.accessToken
  if (!token) {
    console.error('no accessToken in credentials file')
    return
  }

  let usage
  try {
    const body = await request('https://api.anthropic.com/api/oauth/usage', {
      headers: {
        'Authorization': `Bearer ${token}`,
        'anthropic-beta': 'oauth-2025-04-20',
        'User-Agent': 'claude-code/2.1.170',
      },
    })
    usage = JSON.parse(body)
  } catch (e) {
    console.error('failed to fetch usage:', e.message)
    return
  }

  const fh = usage.five_hour || {}
  const sd = usage.seven_day || {}

  const payload = JSON.stringify({
    member_name: memberName,
    five_hour_utilization: fh.utilization ?? null,
    five_hour_resets_at: fh.resets_at ?? null,
    seven_day_utilization: sd.utilization ?? null,
    seven_day_resets_at: sd.resets_at ?? null,
  })

  try {
    await request(SUPABASE_EDGE_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Content-Length': Buffer.byteLength(payload),
      },
    }, payload)
    console.log(`usage pushed for ${memberName}`)
  } catch (e) {
    console.error('failed to push:', e.message)
  }
}

main()
